mongodbatlas/resource_mongodbatlas_cluster_test.go

```diff
@@ -726,3 +826,138 @@ func testAccMongoDBAtlasClusterAWSConfigdWithLabels(projectID, name, backupEnabl
+			provider_name = "%[4]s"
+			region        = "%[5]s"
+		}
+
+		resource "aws_vpc_endpoint" "ptfe_service" {
+			vpc_id             = "%[6]s"
+			service_name       = "${mongodbatlas_private_endpoint.test.endpoint_service_name}"
+			vpc_endpoint_type  = "Interface"
+			subnet_ids         = ["%[7]s"]
+			security_group_ids = ["%[8]s"]
+		}
+
+		resource "mongodbatlas_private_endpoint_interface_link" "test" {
+			project_id            = "${mongodbatlas_private_endpoint.test.project_id}"
+			private_link_id       = "${mongodbatlas_private_endpoint.test.private_link_id}"
+			interface_endpoint_id = "${aws_vpc_endpoint.ptfe_service.id}"
+		}
+
+		resource "mongodbatlas_cluster" "test" {
+		  project_id             = "%[3]s"
+		  name                   = "%[9]s"
+		  disk_size_gb           = 5
+
+		  //Provider Settings "block"
+		  provider_name               = "AWS"
+		  provider_region_name        = "${upper(replace("%[5]s", "-", "_"))}"
+		  provider_instance_size_name = "M10"
+		  provider_backup_enabled     = true // enable cloud provider snapshots
+		  provider_disk_iops          = 100
+		  provider_encrypt_ebs_volume = false
+		  depends_on = ["mongodbatlas_private_endpoint_interface_link.test"]
+		}
+	`, awsAccessKey, awsSecretKey, projectID, providerName, region, vpcID, subnetID, securityGroupID, clusterName)
+}
+
+func testAccMongoDBAtlasClusterConfigAzureWithNetworkPeering(projectID, providerName, directoryID, subcrptionID, resourceGroupName, vNetName, clusterName, atlasCidrBlock, region string) string {
+	return fmt.Sprintf(`
+		resource "mongodbatlas_network_container" "test" {
+			project_id       = "%[1]s"
+			atlas_cidr_block = "%[8]s"
+			provider_name    = "%[2]s"
+			region           = "%[9]s"
+		}
+
+		resource "mongodbatlas_network_peering" "test" {
+			project_id            = "%[1]s"
+			atlas_cidr_block      = "192.168.0.0/21"
+			container_id          = mongodbatlas_network_container.test.container_id
+			provider_name         = "%[2]s"
+			azure_directory_id    = "%[3]s"
+			azure_subscription_id = "%[4]s"
+			resource_group_name   = "%[5]s"
+			vnet_name             = "%[6]s"
+		}
+
+		resource "mongodbatlas_cluster" "test" {
+			project_id   = "%[1]s"
+			name         = "%[7]s"
+
+			replication_factor           = 3
+			auto_scaling_disk_gb_enabled = true
+			mongo_db_major_version       = "4.0"
+
+			//Provider Settings "block"
+			provider_name               = "%[2]s"
+			provider_disk_type_name     = "P6"
+			provider_instance_size_name = "M10"
+			provider_region_name        = "%[9]s"
+
+			depends_on = ["mongodbatlas_network_peering.test"]
+		}
+	`, projectID, providerName, directoryID, subcrptionID, resourceGroupName, vNetName, clusterName, atlasCidrBlock, region)
+}
+
+func testAccMongoDBAtlasClusterConfigGCPWithNetworkPeering(projectID, providerName, gcpProjectID, gcpPeeringName, clusterName string) string {
+	return fmt.Sprintf(`
+		provider "google" {
+			project     = "%[3]s"
+			region      = "us-east4"
```

Comment:
another good one to just make into a param?

Reply:
Thank you so much, I will change that :)